simulation/traffic_simulator/src/hdmap_utils/hdmap_utils.cpp

```diff
@@ -796,19 +796,20 @@ auto HdMapUtils::getPreviousLanelets(const lanelet::Id lanelet_id, const double
 {
   lanelet::Ids ret;
   double total_distance = 0.0;
-  ret.push_back(lanelet_id);
+  lanelet::Id current_lanelet_id = lanelet_id;
```

Comment:
dirty == untested, but try to do something like this (especially names like `else_ids` are unclear)
```
auto HdMapUtils::getPreviousLanelets(const lanelet::Id current_lanelet_id, const double distance) const -> lanelet::Ids
{
  lanelet::Ids previous_lanelets_ids;
  double total_distance = 0.0;
  previous_lanelets_ids.push_back(current_lanelet_id);
  while (total_distance < distance) {
    const auto & reference_lanelet_id = previous_lanelets_ids.back();
    if (const auto straight_lanelet_ids =
          lanelet_map::previousLaneletIds(reference_lanelet_id, "straight");
        !straight_lanelet_ids.empty()) {
      total_distance = total_distance + lanelet_map::laneletLength(straight_lanelet_ids[0]);
      previous_lanelets_ids.push_back(straight_lanelet_ids[0]);
    } else if (auto non_straight_lanelet_ids =
                 lanelet_map::previousLaneletIds(reference_lanelet_id);
               !non_straight_lanelet_ids.empty()) {
      total_distance = total_distance + lanelet_map::laneletLength(non_straight_lanelet_ids[0]);
      previous_lanelets_ids.push_back(non_straight_lanelet_ids[0]);
    } else {
      break;
    }
  }
  return previous_lanelets_ids;
}
```

Reply:
done, but lanelet_map::* functions changed to previously used functions